File: PolicyPlus/AdmxStructures.vb
' Raw data loaded from ADMX files
Public Class AdmxProduct
    Public ID As String
    Public DisplayCode As String
    Public Type As AdmxProductType
    Public Version As Integer
    Public Parent As AdmxProduct
    Public DefinedIn As AdmxFile
End Class
Public Enum AdmxProductType
    Product
    MajorRevision
    MinorRevision
End Enum
Public Class AdmxSupportDefinition
    Public ID As String
    Public DisplayCode As String
    Public Logic As AdmxSupportLogicType
    Public Entries As List(Of AdmxSupportEntry)
    Public DefinedIn As AdmxFile
End Class
Public Enum AdmxSupportLogicType
    Blank
    AllOf
    AnyOf
End Enum
Public Class AdmxSupportEntry
    Public ProductID As String
    Public IsRange As Boolean
    Public MinVersion As Integer?
    Public MaxVersion As Integer?
End Class
Public Class AdmxCategory
    Public ID As String
    Public DisplayCode As String
    Public ExplainCode As String
    Public ParentID As String
    Public DefinedIn As AdmxFile
End Class
Public Class AdmxPolicy
    Public ID As String
    Public Section As AdmxPolicySection
    Public CategoryID As String
    Public DisplayCode As String
    Public ExplainCode As String
    Public SupportedCode As String
    Public PresentationID As String
    Public ClientExtension As String
    Public RegistryKey As String
    Public RegistryValue As String
    Public AffectedValues As PolicyRegistryList
    Public Elements As List(Of PolicyElement)
    Public DefinedIn As AdmxFile
End Class
Public Enum AdmxPolicySection
    Machine = 1
    User = 2
    Both = 3
End Enum

File: PolicyPlus/AdmxStructures.cs
using System.Collections.Generic;

namespace PolicyPlus
{
    // Raw data loaded from ADMX files
    public class AdmxProduct
    {
        public string ID;
        public string DisplayCode;
        public AdmxProductType Type;
        public int Version;
        public AdmxProduct Parent;
        public AdmxFile DefinedIn;
    }

    public enum AdmxProductType
    {
        Product,
        MajorRevision,
        MinorRevision
    }

    public class AdmxSupportDefinition
    {
        public string ID;
        public string DisplayCode;
        public AdmxSupportLogicType Logic;
        public List<AdmxSupportEntry> Entries;
        public AdmxFile DefinedIn;
    }

    public enum AdmxSupportLogicType
    {
        Blank,
        AllOf,
        AnyOf
    }

    public class AdmxSupportEntry
    {
        public string ProductID;
        public bool IsRange;
        public int? MinVersion;
        public int? MaxVersion;
    }

    public class AdmxCategory
    {
        public string ID;
        public string DisplayCode;
        public string ExplainCode;
        public string ParentID;
        public AdmxFile DefinedIn;
    }

    public class AdmxPolicy
    {
        public string ID;
        public AdmxPolicySection Section;
        public string CategoryID;
        public string DisplayCode;
        public string ExplainCode;
        public string SupportedCode;
        public string PresentationID;
        public string ClientExtension;
        public string RegistryKey;
        public string RegistryValue;
        public PolicyRegistryList AffectedValues;
        public List<PolicyElement> Elements;
        public AdmxFile DefinedIn;
    }

    public enum AdmxPolicySection
    {
        Machine = 1,
        User = 2,
        Both = 3
    }
}